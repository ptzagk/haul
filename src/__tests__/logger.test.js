/**
 * Copyright 2017-present, Callstack.
 * All rights reserved.
 *
 * @flow
 */

import stripAnsi from 'strip-ansi';
import logger from '../logger';

beforeEach(() => {
  // $FlowFixMe
  console.log = jest.fn();
});
afterEach(() => jest.clearAllMocks());

test('section logs are always separated by one newline', () => {
  logger.info('single line');
  logger.info('multiple', 'entries');
  logger.info('single line', 'with a newline\n');

  logger.warn('single line');
  logger.warn('multiple', 'entries');
  logger.warn('single line', 'with a newline\n');

  logger.error('single line');
  logger.error('multiple', 'entries');
  logger.error('single line', 'with a newline\n');

  logger.done('single line');
  logger.done('multiple', 'entries');
  logger.done('single line', 'with a newline\n');

  const { calls } = console.log.mock;
  const logs = calls.map(call => call.join(' ')).join('\n');
  expect(stripAnsi(logs)).toMatchSnapshot();
});

test('debug logs are displayed without extra newlines', () => {
  logger.debug('pigeon', 'no space above');
  logger.debug('pigeon', 'no space above, but has newline\n');
  logger.debug('pigeon', 'space above');
  const { calls } = console.log.mock;
  const logs = calls.map(call => call.join(' ')).join('\n');
  expect(stripAnsi(logs)).toMatchSnapshot();
});
