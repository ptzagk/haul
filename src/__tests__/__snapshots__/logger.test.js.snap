// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`debug logs are displayed without extra newlines 1`] = `
"PIGEON no space above
PIGEON no space above, but has newline

PIGEON space above"
`;

exports[`section logs are always separated by one newline 1`] = `
"
 INFO  single line

 INFO  multiple entries

 INFO  single line with a newline

 WARN  single line

 WARN  multiple entries

 WARN  single line with a newline

 ERROR  single line

 ERROR  multiple entries

 ERROR  single line with a newline

 DONE  single line

 DONE  multiple entries

 DONE  single line with a newline
"
`;
